### PR TITLE
Implement checkCloudSdk task

### LINKED ISSUE #212: Split SDK installation and SDK validation into two separate tasks
Either a `downloadCloudSdk` or `checkCloudSdk` task should run depending on the tools extension parameters that are specified in build.gradle, rather than having each handled by the same task.

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #212

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFactory) {
+    this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
+  }
+
+  /** Task entrypoint : Download/update/verify Cloud SDK installation. */
+  @TaskAction
+  public void checkCloudSdkAction() {
+    File home = toolsExtension.getCloudSdkHome();
+    if (home == null) {
+      throw new GradleException("SDK home directory must be specified for validation.");
```
Comment (TadCordle) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: This task never runs if `cloudSdkHome` is set in build.gradle, so this should only happen if the user explicitly calls the task without configuring a home directory. Do we want the user to be allowed to run these tasks themselves, or should they only be automatically called?

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  public void setToolsExtension(ToolsExtension toolsExtension) {
+    this.toolsExtension = toolsExtension;
+  }
+
+  public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFactory) {
+    this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
+  }
+
+  /** Task entrypoint : Download/update/verify Cloud SDK installation. */
```
Comment (coollog) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: Just `verify`

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java (hunk 1 above)
Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: home == null should never occur should it?

Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: oh I see what you're saying.

File: src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  @TaskAction
+  public void checkCloudSdkAction() {
+    File home = toolsExtension.getCloudSdkHome();
+    if (home == null) {
+      throw new GradleException("SDK home directory must be specified for validation.");
+    }
+
+    String version = toolsExtension.getCloudSdkVersion();
+    CloudSdk cloudSdk = cloudSdkBuilderFactory.newBuilder().build();
+    if (!version.equals("LATEST") && !version.equals(cloudSdk.getVersion().toString())) {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: I don't think LATEST is an actual option for the cloud sdk version. Only the second check is necessary.

Comment (TadCordle) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: but LATEST is an option for the tools extension. If I only do the version.equals() check, it'll fail if the user ever specifies LATEST in their build.gradle.

Comment (loosebazooka) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: Oh I see, I didn't realize a user could specify latest. I don't know if we should include LATEST as an option the user can enter, if they don't specify a version, isn't it implied latest?

Comment (TadCordle) [reply src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java]: Hmm yeah, that is true. I guess I can remove the LATEST option.

**Comment by loosebazooka** [comment]
Can you add a test in "AppEngine<Standard/Flexible>PluginTest"  to make sure this task is included when a cloud SDK home is specified and the download task is not?

**Pull request merged**